repo/migrations/Migration012.go

```diff
@@ -0,0 +1,261 @@
+	AverageRating      float32  `json:"averageRating"`
+	RatingCount        uint32   `json:"ratingCount"`
+	ModeratorIDs       []string `json:"moderators"`
+	AcceptedCurrencies []string `json:"acceptedCurrencies"`
+	CoinType           string   `json:"coinType"`
+}
+
+func Migration012_listingHasNewFeaturesAndOldVersion(sl *pb.SignedListing) bool {
+	metadata := sl.Listing.Metadata
+	if metadata == nil {
+		return false
+	}
+	return metadata.Version == 3 &&
+		metadata.PriceModifier != 0
+}
+
+func Migration012_GetIdentityKey(repoPath, databasePassword string, testnetEnabled bool) ([]byte, error) {
+	db, err := OpenDB(repoPath, databasePassword, testnetEnabled)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+
+	var identityKey []byte
+	err = db.
+		QueryRow("select value from config where key=?", "identityKey").
+		Scan(&identityKey)
+	if err != nil {
+		return nil, err
+	}
+	return identityKey, nil
+}
+
+func (Migration012) Up(repoPath, databasePassword string, testnetEnabled bool) error {
+	listingsIndexFilePath := path.Join(repoPath, "root", "listings.json")
+
+	// Find all crypto listings
+	listingsIndexJSONBytes, err := ioutil.ReadFile(listingsIndexFilePath)
+	if err != nil {
+		return err
+	}
+
+	listingsIndex := []*Migration012_ListingData{}
+	err = json.Unmarshal(listingsIndexJSONBytes, &listingsIndex)
+	if err != nil {
+		return err
+	}
+
+	cryptoListings := []*Migration012_ListingData{}
+	for _, listingAbstract := range listingsIndex {
+		if listingAbstract.ContractType == "CRYPTOCURRENCY" {
+			cryptoListings = append(cryptoListings, listingAbstract)
+		}
+	}
+
+	// Finish early If no crypto listings
+	if len(cryptoListings) == 0 {
+		return writeRepoVer(repoPath, 13)
+	}
+
+	// Check each crypto listing for markup
+	markupListings := []*pb.SignedListing{}
+	for _, listingAbstract := range cryptoListings {
+		listingJSONBytes, err := ioutil.ReadFile(migration012_listingFilePath(repoPath, listingAbstract.Slug))
+		if err != nil {
+			return err
+		}
+		sl := new(pb.SignedListing)
+		err = jsonpb.UnmarshalString(string(listingJSONBytes), sl)
+		if err != nil {
+			return err
+		}
+
+		if Migration012_listingHasNewFeaturesAndOldVersion(sl) {
+			markupListings = append(markupListings, sl)
+		}
+	}
+
+	// Finish early If no crypto listings with new features are found
+	if len(markupListings) == 0 {
+		return writeRepoVer(repoPath, 13)
+	}
+
+	// Setup signing capabilities
+	identityKey, err := Migration012_GetIdentityKey(repoPath, databasePassword, testnetEnabled)
+	if err != nil {
+		return err
+	}
+
+	identity, err := ipfs.IdentityFromKey(identityKey)
+	if err != nil {
+		return err
+	}
+
+	// IPFS node setup
+	r, err := fsrepo.Open(repoPath)
+	if err != nil {
+		return err
+	}
+	cctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cfg, err := r.Config()
+	if err != nil {
+		return err
+	}
+
+	cfg.Identity = identity
+
+	ncfg := &ipfscore.BuildCfg{
+		Repo:   r,
+		Online: true,
```

Comment:
Why don't you set this to  `Online: false`. Since this is not configuring Tor it would someone's privacy if it tries to make network connections.

Reply:
Good call. I was trying to stay as close the the example on cmd/convert.go but overlooked the Online flag when looking for things that should be changed.